ambry-router/src/main/java/com/github/ambry/router/RouterUtils.java

```diff
@@ -245,4 +257,30 @@ static Port getPortToConnectTo(ReplicaId replicaId, boolean routerEnableHttp2Net
      */
     T readFrom(DataInputStream stream) throws IOException;
   }
+
+  /**
+   * Used to map sent {@link Response} objects to received {@link Response} objects. For example, in {@link GetResponse},
+   * the message content is held by {@code GetResponse#toSend} object at the sender side while it is held and accessed
+   * as {@code GetResponse#stream} at the receiver side. Hence, we need to construct new {@link Response} objects before
+   * using them. This method is applicable when sending and receiving responses as java objects (instead of serialized
+   * bytes) in the same process via {@link LocalNetworkClient}.
+   * At the moment, other responses like {@link PutResponse}, {@link DeleteResponse}, etc don't need any special
+   * handling and can be referenced as they are.
+   * @param sentResponse {@link Response} object constructed at sender side.
+   * @return {@link Response} object constructed at receiver side.
+   */
+  static Response mapToReceivedResponse(Response sentResponse) {
+    Response receivedResponse;
+    if (sentResponse instanceof GetResponse) {
+      GetResponse getResponse = (GetResponse) sentResponse;
+      // Not setting releaseOnClose to true when creating ByteBufInputStream since the underlying bytebuf is released
+      // when the Response object is closed as part of {@code ResponseInfo#release}.
+      receivedResponse = new GetResponse(getResponse.getCorrelationId(), getResponse.getClientId(),
```

Comment:
What would be the level of effort to build the functionality to expose `toSend.content()` and `blob.content()` as an input stream directly (by lazily constructing it) into the `PutRequest` and `GetResponse` objects? This code looks like it could be error prone if other constructors get added or other types of requests/responses need this logic.

Alternatively, what does it take to be able to stop using the `getInputStream` method in these cases?